benchmarks/async_alloc/async_alloc.cpp

```diff
@@ -0,0 +1,104 @@
+#include <iostream>
+#include <sstream>
+#include <utility>
+
+// Kokkos Headers
+#include "Kokkos_Core.hpp"
+
+using MemorySpace = Kokkos::DefaultExecutionSpace::memory_space;
+
+// Store the individual timings
+std::vector<std::pair<size_t, double>> inner_loop_times;
+
+// The pair returned are the first_alloc_time and the loop time
+//
+std::pair<double, double> test(bool up) {
+  int iters      = 50;
+  size_t minimum = 8 / sizeof(float);  // 64K
+
+  size_t gb = 1024 * 1024 * 1024 / sizeof(float);  // number of floats per GiB
+  size_t maximum = gb;  // on 32 bit, we make 1GiB the max
+
+  // On 64-bit we make 16 GiB the max
+  if constexpr (sizeof(size_t) == 8) maximum *= 16;
+
+  std::vector<size_t> sizes;
+  if (up) {
+    for (size_t num = minimum; num <= maximum; num *= 2) sizes.push_back(num);
+  } else {
+    for (size_t num = maximum; num >= minimum; num /= 2) sizes.push_back(num);
+  }
+
+  Kokkos::Timer first_alloc_timer;
+  {  // Prime the pump - first long alloc -- Time it.
+    Kokkos::View<float *, MemorySpace> dummy("unlabeled", 64);
```

Comment:
The size doesn't matter, yes I went for 256K here, but it is not important. I just want one first call that will call the allocator. Then it will set the poolsize. As you see the view is then immediately freed.